src/Oro/Bundle/LocaleBundle/DoctrineExtensions/DBAL/Types/UTCDateTimeType.php

```diff
@@ -40,6 +40,11 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
             (self::$utc) ? self::$utc : (self::$utc = new \DateTimeZone('UTC'))
         );
 
+        // Dates are stored in UTC timezone
+        // @see PIM-4216
+        $serverTimezone = date_default_timezone_get();
+        $val->setTimezone(new \DateTimeZone($serverTimezone));
```

Comment:
Not a good idea to do that directly. Lots of possible impact, as updated and created date are properly displayed without this fix, meaning that a transformation is already done. Doing it twice will create new problems.